src/proteus/dswx_hls.py

```diff
@@ -1772,14 +1810,46 @@ def _load_hls_from_file(filename, image_dict, offset_dict, scale_dict,
 
     metadata = layer_gdal_dataset.GetMetadata()
 
+    # read image
+    if flag_debug:
+        logger.info('reading in debug mode')
+        image = layer_gdal_dataset.ReadAsArray(
+            xoff=0, yoff=0, xsize=1000, ysize=1000)
+    else:
+        image = layer_gdal_dataset.ReadAsArray()
+
+
+    # read `fill_value`
+    if fill_value is None and '_FillValue' in metadata.keys():
+        fill_value = float(metadata['_FillValue'])
+    elif fill_value is None:
+        fill_value = -9999
+
+    # create/update `invalid_ind_array`
+    # (invalid pixels are cumulative)
+    if 'invalid_ind_array' not in image_dict.keys():
+        invalid_ind_array = image == fill_value
+    else:
+        invalid_ind_array = np.logical_or(image_dict['invalid_ind_array'],
+                                          image == fill_value)
+
+    image_dict['invalid_ind_array'] = invalid_ind_array
```

Comment:
Sanity Check: Have you tested one of the problem granules that was exhibiting the "railroad tracks" with this code? Meaning, have you visually inspected outputs to ensure that the new process for generating the invalid indices is working correctly and is correctly being applied to all layers? Thanks!

Reply:
Not yet, if you could help me with that that would be great.